editor/store/reducer.js

```diff
@@ -673,9 +673,12 @@ export function blockSelection( state = {
 				return state;
 			}
 
-			// If there is replacement block(s), assign first's UID as the next
-			// selected block. If empty replacement, reset to null.
-			const nextSelectedBlockUID = get( action.blocks, [ 0, 'uid' ], null );
+			// If blockToSelectUid is set assign it as the next selected block.
+			// Otherwise, If there is replacement block(s), assign first's UID
+			// as the next selected block. If empty replacement, reset to null.
+			const nextSelectedBlockUID = action.blockToSelectUid ?
+				action.blockToSelectUid :
+				get( action.blocks, [ 0, 'uid' ], null );
 
 			return {
 				...state,
```

Comment:
I know it was already the case before but I feel the `REPLACE_BLOCKS` action should not touch the selection at all unless the currently selected block is being replaced.